commit 684f067dde3b798877655cdda4eab8d7c26b2510
Author: George Koehler <kernigh@gmail.com>
Date:   Thu Oct 3 20:06:38 2019 -0400

    OpenBSD is missing SI_ASYNCIO and SI_MESGQ
    
    Check if SI_ASYNCIO and SI_MESGQ are defined as macros.  This allows
    to run tests on OpenBSD 6.5, where the macros are missing.
    
    This is identical to patch-boost_test_impl_execution_monitor_ipp in
    OpenBSD Ports, except that I added a comment.

    Can be removed when we use Boost 1.72.x or later.

diff --git a/boost/test/impl/execution_monitor.ipp b/boost/test/impl/execution_monitor.ipp
index ccc44972..77a01e21 100644
--- a/boost/test/impl/execution_monitor.ipp
+++ b/boost/test/impl/execution_monitor.ipp
@@ -391,14 +391,19 @@ system_signal_exception::report() const
         report_error( execution_exception::system_error,
                       "signal: the expiration of a timer set by timer_settimer()" );
         break;
+// OpenBSD was missing SI_ASYNCIO and SI_MESGQ
+#ifdef SI_ASYNCIO
     case SI_ASYNCIO:
         report_error( execution_exception::system_error,
                       "signal: generated by the completion of an asynchronous I/O request" );
         break;
+#endif
+#ifdef SI_MESGQ
     case SI_MESGQ:
         report_error( execution_exception::system_error,
                       "signal: generated by the the arrival of a message on an empty message queue" );
         break;
+#endif
     default:
         break;
     }
